### ISSUE TITLE
ValidateWithMethod not working when specified via XML in OVal 2.x

### ISSUE THREAD

**Issue by bharattrikha03** [sebthom/oval#65]
Hi,
Recently i have updated my application to JDK 8 and have also updated the Oval Api from 1.85 to 2.1.0 . After upgrading all the validations are working properly except for **ValidateWithMethod**  . As checked in the class ValidateWithMethodCheck 

```java
private final ConcurrentMap<Class<?>, Method> validationMethodsByClass = Validator.getCollectionFactory().createConcurrentMap();
```

Here validationMethodsByClass is coming as Null due to which code is breaking in the below method

```java
   @Override
   public boolean isSatisfied(final Object validatedObject, final Object valueToValidate, final OValContext context, final Validator validator)
      throws ReflectionException {
      if (valueToValidate == null && ignoreIfNull)
         return true;

      final Class<?> clazz = validatedObject.getClass();
      Method method = validationMethodsByClass.get(clazz);
      if (method == null) {
         method = ReflectionUtils.getMethodRecursive(clazz, methodName, parameterType);
         validationMethodsByClass.put(clazz, method);
      }

      if (method == null)
         throw new InvalidConfigurationException("Method " + clazz.getName() + "." + methodName + "(" + parameterType + ") not found. Is [" + parameterType
            + "] the correct value for [@ValidateWithMethod.parameterType]?");
      // explicit cast to workaround:
      // "type parameters of <T>T cannot be determined; no unique maximal instance exists for type variable T with upper bounds boolean,java.lang.Object"
      return (Boolean) ReflectionUtils.invokeMethod(method, validatedObject, valueToValidate);
   }
```

Below configuration I am using :

To configure ovalValidator :

```java
package com.aon.salesforce.validator.oval.constraint;

import java.io.File;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.thoughtworks.xstream.XStream;

import net.sf.oval.Validator;
import net.sf.oval.configuration.xml.XMLConfigurer;

@Configuration
public class MyOvalConfig {

	@Bean
	  public Validator ovalValidator(@Value("${prprConfigRoot}") String prprConfigRoot) {
	    XStream xstream = XMLConfigurer.createXStream();
		xstream.allowTypesByWildcard(new String[] {
	      "com.aon.salesforce.validator.oval.constraint.*"
	    });
		XMLConfigurer xmlConfig = new XMLConfigurer(
				xstream, 
	      new File(prprConfigRoot + "/databinder/OvalXMLConfiguration.xml")
	    );
	    return new Validator(xmlConfig);
	  }
}
```

xml file where ValidatewithMethd is configured :

```xml
		<field name="countries" overwrite="false">
			<assertConstraintSet id="semicolon.deliminated" />
			<validateWithMethod methodName="validateMaxLimit"
				parameterType="java.lang.String" message="Countries should not be more than 100"
				errorCode="{excel-field-name}" />
		</field>
```

Please suggest

**Comment by sebthom** [comment]
Fixed in OVal 2.1.6. Thanks for reporting.

**Comment by bharattrikha03** [comment]
Thanks for the fix :)
